### ISSUE TITLE
Protocol Handler: WEBM does not load when used in video HTML tag

### ISSUE THREAD

**Issue by lidel** [mozilla/libdweb#57]
### Summary

WEBM video fails to load on initial load when used in `<video>` tag.
If opened directly at least once, then loads fine.

Discovered this problem in IPFS Companion, which was streaming videos from IPFS,
but I created a simplified repro without IPFS that downloads file via HTTP and produces the same error.
Tested with 2MB and 22MB files, same result for both. Steps below use 2MB file.

### How to Reproduce

1. Run `demo:protocol` from my `video-render-bug-demo` branch:
    ```
    git clone -b video-render-bug-demo --depth 1 https://github.com/lidel/libdweb.git
    cd libdweb
    yarn && yarn demo:protocol
    ```
1. Open `dweb://html/` first: loading via `<video>` tag will fail with [this error](https://user-images.githubusercontent.com/157609/43795565-083fec06-9a82-11e8-8333-6a4852ecc847.png).
    - Looks like a race-condition of sorts related to content-type sniffing. It failed to guess content-type when video was not yet in cache. @Gozala  this may be remotely related to discussion about content-type we had at DWeb: if browser could wait until API returns content-encoding, content-type along with content's asynciterator, that could solve this issue as well.

1. Open `dweb://html/video.webm` directly: it will [load and play just fine](https://user-images.githubusercontent.com/157609/43795624-38e853de-9a82-11e8-81ef-0fdda285eba8.png).

1. Open `dweb://html/` again: loading via `<video>` tag [will work this time](https://user-images.githubusercontent.com/157609/43795672-5cd60908-9a82-11e8-9555-6a2e3590a76b.png).
    - It seems browser's content-type sniffing fails when video is not yet in cache, but works fine after initial load.

1. Open `dweb://foo/`: loading via `<video>` tag _from different origin_ [will fail](https://user-images.githubusercontent.com/157609/43795827-cb2d4a42-9a82-11e8-8d26-9a6e85318100.png).
   -  This is basically the same issue as in step 4) in #54, so I guess we will follow it in https://github.com/mozilla/libdweb/issues/55

**Comment by Gozala** [comment]
If I had to guess it's the same as #54 as `<img />`  accepts `image/*` likely `<video/> accepts `video/*` but we default to `application/x-unknown-content-type`. I'll verify if that is the case and if so will close this as duplicate for #54

**Comment by Gozala** [comment]
Well only after running it myself I did noticed that warning in the console actually explicitly mentioned it could not match content type.

**Comment by Gozala** [comment]
As I suspected it's the same issue as #54 as this change loads the video

```diff
diff --git a/demo/protocol/protocol.js b/demo/protocol/protocol.js
index 4f88087..8270bf3 100644
--- a/demo/protocol/protocol.js
+++ b/demo/protocol/protocol.js
@@ -50,6 +50,7 @@ browser.protocol.registerProtocol("dweb", request => {
     }
     case "dweb://html/video.webm": {
       return {
+        contentType: "video/webm",
         content: (async function*() {
           const response = await fetch(
             "https://ipfs.io/ipfs/bafybeigwa5rlpq42cj3arbw27aprhjezhimhqkvhbb2kztjtdxyhjalr3q/big-buck-bunny_trailer.webm",
```

I'm closing this as being duplicate of #54

**Comment by Gozala** [comment]
Reopening because #60 fixed #54 without fixing this one.

**Comment by lidel** [comment]
Just for the record, retested it with async API and works as expected, thanks!
(cross-origin fails but that is separate issue tracked in https://github.com/mozilla/libdweb/issues/55)

![screenshot](https://user-images.githubusercontent.com/157609/44027096-895cddea-9ef5-11e8-9628-c7ba3cfba068.png)
